### ISSUE TITLE
navigation.setParams() hide keyboard

### ISSUE THREAD

**Issue by HsuWai** [react-navigation/react-navigation#4677]
Software keyboard on Android is hidden after calling this.props.navigation.setParams. This makes impossible to input text. 

react-navigation : 2.6.2
react-native : 0.55.3

The following is my code: 

                  
            import React, { Component } from "react";
            import {
                View,
                Text,
                TextInput,
                Platform,
                TouchableOpacity,
                ActivityIndicator,
                Alert,
                Keyboard
            } from "react-native";
            import { connect } from "react-redux";
            import _ from "lodash";
            import IoniIcon from "react-native-vector-icons/Ionicons";
            import EntypoIcon from "react-native-vector-icons/Entypo";
            import MaterialIcon from "react-native-vector-icons/MaterialIcons";

            class MyScreen extends Component {
                static navigationOptions = ({ navigation }) => {
                    const { state } = navigation;
                    return {
                        title: state.params ? state.params.room_name : "Detail",
                        headerTitleStyle: styles.headerTitleStyle,
                        headerLeft: (
                            <TouchableOpacity onPress={() => state.params.handleGoBack()}>
                                <View style={styles.backIconStyle}>
                                    <IoniIcon
                                        name="md-arrow-round-back"
                                        size={Metrics.icons.small}
                                        color="black"
                                    />
                                </View>
                            </TouchableOpacity>
                        ),
                        headerRight:
                        state.params.admin && state.params.update ? (
                            <TouchableOpacity onPress={() => state.params.handleUpdateGroup()}>
                                <View>
                                    <EntypoIcon
                                        name="check"
                                        size={Metrics.icons.ssmall}
                                        style={{ marginRight: 10 }}
                                        color="black"
                                    />
                                </View>
                            </TouchableOpacity>
                        ) : (
                                <View />
                            ),
                        headerStyle: {
                            backgroundColor: Colors.headerGray,
                            elevation: 0 //hide navbar line
                        },
                        headerTintColor: Colors.text
                    };
                };

                constructor(props) {
                    super(props);
                    this.state = {
                        groupName: "",
                    };
                }

                componentDidMount() {
                    this.props.navigation.setParams({
                        handleGoBack: this.handleGoBack.bind(this),
                        handleUpdateGroup: this.handleUpdateGroup.bind(this),
                        update: false
                    });
                }
                handleGoBack() {
                    this.props.navigation.goBack();
                }
                handleUpdateGroup() {
                }

                changeGroupName = groupName => {
                    this.setState({ groupName });
                };

                onFocusText = () => {
                    this.props.navigation.setParams({
                        update: true
                    });
                };

                checkGroupName = value => {
                };

                render() {
                    const { navigation } = this.props;
                    return (
                        <View
                            style={[
                                styles.createContainer,
                            ]}
                        >
                            <TextInput
                                autoCapitalize="none"
                                style={[styles.newGroupName]}
                                underlineColorAndroid="transparent"
                                placeholder={"GROUP CHAT NAME"}
                                value={this.state.groupName}
                                onChangeText={this.changeGroupName}
                                placeholderTextColor={Colors.iconGrey}
                                onFocus={this.onFocusText}
                                onBlur={() => this.checkGroupName(this.state.groupName)}
                                editable={
                                    navigation.state.params
                                        ? navigation.state.params.admin
                                        : false
                                }
                            />
                        </View>
                    );
                }
            }

            const mapStateToProps = state => {
                return state;
            };

            export default connect(
                mapStateToProps,
                {

                }
            )(MyScreen);

**Comment by kirakik** [comment]
Hi @HsuWai, I [tried replicating this issue](https://snack.expo.io/ryFDIpDmX) but wasn't able to. If you could replicate it in a Snack that would be very helpful.

Is there a particular reason you are setting the functions `handleGoBack` and `handleUpdateGroup` in the navigation params? Same for the `update` property, I don't understand the use case you are trying to cover by setting it to `true` when the text input becomes focused. If you could elaborate, I could help you find an alternative solution.

**Comment by HsuWai** [comment]
Hello @kirakik ,  the behavior I want is to show `update icon` only if the user focus on the TextField. This is the desired behavior from UI Team, so I set navigation params of `update` to true if user focus on textfield. The fuctions `handleGoBack` and `handleUpdateGroup` is needed to call some fuctions. I replicated in Snack and the issue can't be reproduced. But, it still occurs in my device. Now, I discussed with the UI team and change behavior. So, no need to call setParams when the textinput becomes focused. I appreciate for your response, Kirakik.
